internal/tiltfile/tiltfile_state.go

```diff
@@ -608,6 +632,32 @@ func (s *tiltfileState) assembleK8sV2() error {
 		return err
 	}
 
+	unresourcedFragmentsToEntities := k8s.FragmentsToEntities(s.k8sUnresourced)
+	// TODO(dmiller): can we know the size of these?
+	unresourcedFragmentNames := []string{}
+	for name := range unresourcedFragmentsToEntities {
+		unresourcedFragmentNames = append(unresourcedFragmentNames, name)
+	}
+	sort.Sort(byLen(unresourcedFragmentNames))
+	uniqueFragments := map[string]k8s.K8sEntity{}
+	uniqueFragmentNames := []string{}
+	seenEntities := map[k8s.K8sEntity]interface{}{}
+
+	for _, name := range unresourcedFragmentNames {
```

Comment:
...wait why did you put all the fragments from the fragment -> entity map into an array to loop over that instead of just looping over the keys from the map?